### ISSUE TITLE
Inside a parser, how do I "back up" one character?

### ISSUE THREAD

**Issue by deanebarker** [lunet-io/markdig#390]
Assuming I've found my pattern, I also want to include the character before _if it's a space_. So I want to do something like this:

```
if(slice.PeekCharExtra(-1).IsSpace())
{
   slice.PrevChar();   // Note: this does not exist
}
```

I saw the `Start` property on `StringSlice`, but I can't figure out what character index I'm on now, so that didn't seem to help.

**Comment by xoofx** [comment]
> Assuming I've found my pattern, I also want to include the character before if it's a space. So I want to do something like this:

I don't think we are able to handle this case easily. Once a parser has parsed a slice, it will transform it to its internal representation (and we are not fully 100% roundtrip ready, as we don't store all stuffs that a parser could discard e.g trailing spaces)...
So the easiest solution would be to trigger your parser on space and your characters, and match from that instead.
